### PR TITLE
Make the stop endpoint more robust and test

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Handle requesting stop on a non-existent trigger correctly. Add tests for various failure scenarios.

### CODE REVIEW

File: triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala [hunk 1]
```diff
@@ -500,4 +500,42 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
...
+      resp <- Http().singleRequest(req)
+      body <- responseBodyToString(resp)
+      JsObject(fields) = body.parseJson
+      _ <- fields.get("status") should equal(Some(JsNumber(422)))
+      _ <- fields.get("errors") should equal(
+        Some(JsArray(JsString("missing Authorization header with OAuth 2.0 Bearer Token"))))
+    } yield succeed
+  }
+
+  it should "stopping a trigger that can't parse as a UUID gives a 404 response" in withHttpService(
```
Comment (cocreature) [inline triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala:520]: This doesn’t make sense to me. Why does an invalid UUID produce a 404 but a UUID that is not found produces a 422? that seems the wrong way around.

Comment (shayne-fletcher) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala:520]: It's an akka thing. If the path (`pathPrefix("v1" / "stop" / JavaUUID)`) is invalid (the `JavaUUID` fails to parse), it never makes "our" code and a 404 is returned. In the event it is a valid path but the UUID doesn't match any trigger we know we return 422 (unprocessable entity). Make better sense now?

Comment (shayne-fletcher) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala:520]: I could arrange for a 404 in the second case too if you think that's better?

Comment (cocreature) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala:520]: Not really, makes sense that we get a 404 for the invalid UUID but I don’t understand why 422 is a good response here. It is literally a failing lookup. That seems very much like “not found” to me.

Comment (shayne-fletcher) [reply triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala:520]: Ok. Let's go with 404 in both cases.

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 2]
```diff
@@ -373,7 +378,11 @@ object Server {
                     unauthorized =>
                       complete(
                         errorResponse(StatusCodes.UnprocessableEntity, unauthorized.message)),
-                    triggerInstance => complete(successResponse(triggerInstance))
+                    triggerInstance =>
+                      triggerInstance match {
+                        case Some(stoppedTriggerId) => complete(successResponse(stoppedTriggerId))
+                        case None => complete(StatusCodes.NotFound)
```
[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:384]: Well, I don't know. That will result in a DA json response. Do we want a "raw" 404 response consistent with what akka does with an invalid route or, do we want to acknowledge the route and say that the entity it refers to was "not found". I think if the latter then it's not the resource that wasn't found, it's the entity that the resource pertains too that wasn't and unprocessable entity is then the right response (as it were).

[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:384]: What do you think?

[the remark this answers is not in the document]
Comment (cocreature) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:384]: I would go for 404 + JSON response. Then you can differentiate between invalid routes and not found triggers while keeping the (at least to me) expected “not found” response for a trigger that is not found.

[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:384]: Ok, you got it!

[the remark this answers is not in the document]
Comment (rohanjr) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala]: Agreed it should be an `errorResponse` maybe with a message like "no running trigger with id \<uuid\>"

**Pull request merged**